lib/git.zsh

```diff
@@ -12,10 +12,27 @@ function __git_prompt_git() {
 # Outputs current branch info in prompt format
 function git_prompt_info() {
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
-    ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
-    ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) || return 0
-    echo "$ZSH_THEME_GIT_PROMPT_PREFIX${ref#refs/heads/}$(parse_git_dirty)$ZSH_THEME_GIT_PROMPT_SUFFIX"
+  if [[ "$(command git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
+    branches=$(command git branch --list -a 2> /dev/null)
+    if [[ "$branches" == "" && "$?" == "0" ]]; then
+        # Look for initialized folders
+        info="<initialized>"
+    elif branch=$(command git symbolic-ref HEAD 2> /dev/null); then
+        # Branch name in the form "<upstream branch name>/<brach name>"
+        ref=$(command git for-each-ref --format='%(upstream:short)' ${branch} 2> /dev/null) || return 0
+        info=$(command echo "${ref#refs/heads/}")
+        if [[ "${info}" == "" ]]; then
+            # Handle for no remote branch
+            info=$(command echo "<no-remote>/$(command git rev-parse --abbrev-ref HEAD)" 2> /dev/null) || return 0
```

Comment:
Same as before, there's no use using `echo`. Also, I bet there's a way to use only one git call to get this information either with or without a remote branch. `git` calls are expensive!

Reply:
See my comment below: https://github.com/ohmyzsh/ohmyzsh/pull/9188#issuecomment-678034444